roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -33,10 +34,39 @@ class DockerImageAvailability(DockerHostMixin, OpenShiftCheck):
     # we use python-docker-py to check local docker for images, and skopeo
     # to look for images available remotely without waiting to pull them.
     dependencies = ["python-docker-py", "skopeo"]
-    skopeo_img_check_command = "timeout 10 skopeo inspect --tls-verify=false docker://{registry}/{image}"
+    # command for checking if remote registries have an image, without docker pull
+    skopeo_command = "timeout 10 skopeo inspect --tls-verify={tls} {creds} docker://{registry}/{image}"
+    skopeo_example_command = "skopeo inspect [--tls-verify=false] [--creds=<user>:<pass>] docker://<registry>/<image>"
 
     def __init__(self, *args, **kwargs):
         super(DockerImageAvailability, self).__init__(*args, **kwargs)
+
+        self.registries = dict(
+            # set of registries that need to be checked insecurely (note: not accounting for CIDR entries)
+            insecure=set(self.ensure_list("openshift_docker_insecure_registries")),
+            # set of registries that should never be queried even if given in the image
+            blocked=set(self.ensure_list("openshift_docker_blocked_registries")),
+        )
+
+        # ordered list of registries (according to inventory vars) that docker will try for unscoped images
+        regs = self.ensure_list("openshift_docker_additional_registries")
+        # currently one of these registries is added whether the user wants it or not.
+        deployment_type = self.get_var("openshift_deployment_type")
+        if deployment_type == "origin" and "docker.io" not in regs:
+            regs.append("docker.io")
+        elif deployment_type == 'openshift-enterprise' and "registry.access.redhat.com" not in regs:
+            regs.append("registry.access.redhat.com")
+        self.registries["configured"] = regs
+
+        # for the oreg_url registry there may be credentials specified
+        components = self.get_var("oreg_url", default="").split('/')
+        self.registries["oreg"] = "" if len(components) < 3 else components[0]
+        self.skopeo_command_creds = ""
+        oreg_auth_user = self.get_var('oreg_auth_user', default='')
+        oreg_auth_password = self.get_var('oreg_auth_password', default='')
+        if oreg_auth_user != '' and oreg_auth_password != '':
+            self.skopeo_command_creds = "--creds={}:{}".format(quote(oreg_auth_user), quote(oreg_auth_password))
```

Comment:
I'm not sure this will have the desired effect or not with the quoting.

Reply:
```quote_plus`` may be a better option assuming slashes will be problematic and spaces are not allowed. If spaces are allowed and forward slashes are an issue then ``quote`` along with a ``replace()`` should suffice (assuming one is encoding for HTTP transport and not shell escaping).

```
print(string.punctuation,  quote(string.punctuation))
('!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~', '%21%22%23%24%25%26%27%28%29%2A%2B%2C-./%3A%3B%3C%3D%3E%3F%40%5B%5C%5D%5E_%60%7B%7C%7D%7E')

print(string.punctuation,  quote_plus(string.punctuation))
('!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~', '%21%22%23%24%25%26%27%28%29%2A%2B%2C-.%2F%3A%3B%3C%3D%3E%3F%40%5B%5C%5D%5E_%60%7B%7C%7D%7E')
```

Reply:
This is neither shell escaping nor HTTP transport. This is for the ansible `command` module. It doesn't run through the shell, but it still does some kind of lexical parsing to separate out the arguments. For this purpose, putting quotes around things is all we need. I saw other modules in ansible using `pipes.quote` but I have not actually tried this with a registry and a password with special characters like spaces and quotes.

Reply:
```
>>> print pipes.quote("aasdf")
aasdf
>>> print pipes.quote("aa  sdf")
'aa  sdf'
>>> print pipes.quote("aa ' \" sdf")
'aa '"'"' " sdf'
```

Reply:
Yeah, this will probably work.

Reply:
@sosiouxme ah, ok. Thanks for the clarification.